docs/design/multi_slo_doc_support.md

```diff
@@ -0,0 +1,102 @@
+
+## Proposal
+
+### Overview
+
+The 'name' ([example](https://gitlab.cee.redhat.com/service/app-interface/-/blob/8bba50902109207d7e8a0b8f856bec92ede1e482/data/services/ocm/slo-documents/accounts-mgmt.yml#L7)) of these SLO docs will be introduced as an identifier for the SLO metric data stored in dashdotdb.
+
+SLO data will be stored in report-YML files in the following format:
+```
+service_slo:
+  slo_doc_name1:
+  - cluster: app-sre-prod-01
+    namespace: app-interface-production
+    slo_name: Reconciliation time of app-interface integrations
+    slo_value: 27.0
+    slo_target: 95.0
+  slo_doc_name2:
+  - ...
+```
+
+Visual-Qontract will have a new column `Doc Name` in the 'Service SLO' table.
+
+### Schema Changes
+
+Here are three proposals for how a 'slo_doc_name' identifier can be introduced to the schema:
+* [Option 1] (least effort): Update the 'ServiceSLO' table to include a 'SLODocName' column of type 'string'.
+* [Option 2] (author's recommendation): Update the 'ServiceSLO' table to include a 'SLODocName' column of type 'integer', which is a foreign key to a new table 'SLODocName'. Table 'SLODocName' contains 'id' (primary key, integer), and 'name' (string, unique)
+    * Better data normalization that Option 1
+    * Less disruptive to schema than Option 3
+* [Option 3] (most effort): Same as Option 2, and in addition: update all tables that currently have a relationship with table 'ServiceSLO' to instead have a relationship with 'SLODocName'
+
+[This PR](https://github.com/app-sre/dashdotdb/pull/50) uses the "Option 2" approach.
+
+This digram visualizes the proposed schema change, introducing the 'SLODocName' table. (bottom left):
+![](images/multi-slo-doc-schema-change.png)
+
+### Other Changes To DashDotDB 
+
+* dash.db
+  * this needs a "slo_doc_name" property: https://github.com/app-sre/dashdotdb/blob/f88048cd9156afa93a4a0448ecd088c551c8408d/dashdotdb/models/dashdotdb.py
+  * update this to include "slo_doc_name": https://github.com/app-sre/dashdotdb/blob/f88048cd9156afa93a4a0448ecd088c551c8408d/examples/serviceslometrics.json
+  * update this image: https://github.com/app-sre/dashdotdb/blob/f88048cd9156afa93a4a0448ecd088c551c8408d/docs/dashdotdb.png (and the diagram file for it: https://github.com/app-sre/dashdotdb/blob/f88048cd9156afa93a4a0448ecd088c551c8408d/docs/dashdotdb.dia)
+  * Follow the [DB upgrade guide](https://github.com/app-sre/dashdotdb#db-upgrade) to upgrade the schema in Postgres
+
+### Changes To Other Systems
+
+The following are changes to related systems that are expected to need to be made.
+
+* app-interface
+  * add "slo_doc_name": https://gitlab.cee.redhat.com/service/app-interface/-/blob/15d1ce504a9233a04d6119e8d57e58a3bcc31cb0/graphql-schemas/schema.yml#L1916-1922
```

Comment:
Why is this needed?  The SLODocument_v1 already has a name field here.  The SLO document names are in the SLODocumentSLO_v1 schema.

Reply:
Thank you for pointing this out. It is not needed.

I had intended to delete this point.